lib/ui/common/components/cobble_tile.dart

```diff
@@ -332,14 +332,16 @@ class CobbleTile extends StatelessWidget {
 
   /// Will change IconData or ImageProvider to Widget
   static Widget _leadingToWidget(Object? leading, {double size = 25}) {
-    assert(leading == null || leading is IconData || leading is ImageProvider);
+    assert(leading == null || leading is IconData || leading is ImageProvider || leading is Widget);
     if (leading is IconData && leading == reservedIconSpace)
       return SizedBox(width: size + 16.0);
     if (leading is IconData && leading != reservedIconSpace)
       return Icon(
         leading,
         size: size,
       );
+    if (leading is Widget)
```

Comment:
I think this method should be more strict in case of a widget. It should at least limit the widget to size 25x25

Reply:
It actually is 48x48 here, so I'm not sure that's a good idea

Reply:
Yes, I was too hasty, I should have said "limit it to size passed to `_leadingToWidget`". Now that I took second look at it, I'd even suggest renaming `CobbleTile.app` to `CobbleTile.biggerIcon` and reusing that.

Reply:
I think it's appropriate to clarify why I'm more strict about this size. CobbleTile is meant to be reusable component with strict variations. You could have smaller tile, bigger tile, tile with settings etc. and they should be easily to use and difficult to mess up. Ability to pass in arbitrary widget without any check opens up possibility to mess up styling.

Now, when I was designing API for CobbleTile, designs weren't complete yet and I implemented what I saw in designs. Now that there's yet another use case, it makes sense to iterate CobbleTile API but not too much.

We could be even more restrictive in that iteration and allow only parameters `IconData`, `CompIcon` or `ImageProvider `. This would support new use case while still limiting developers.

Reply:
I opted for adding BackCompIcon as a type and use that inside CobbleTile, that seemed like the least painful way to implement this with support for background colour without using the existing intent, which does more than necessary in this case. I did end up making that configurable, since I see we will need size 56 and size 32 in some places